Cargo.toml

```diff
@@ -45,7 +45,7 @@ edition = "2021"
 homepage = "https://astarte.cloud/"
 license = "Apache-2.0"
 repository = "https://github.com/astarte-platform/astarte-device-sdk-rust"
-rust-version = "1.72.0"
+rust-version = "1.78.0"
```

Comment:
I would check if we can remove the transitive and pinned dependencies

Reply:
Removed the transitive dependencies and removed `once-cell`